main.go

```diff
@@ -153,11 +198,25 @@ func main() {
 		}
 	}
 
-	signalCh := make(chan os.Signal, 1)
-	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
-	<-signalCh
+	if *enableLeaderElection {
```

Comment:
Should this be moved up to be right after the elector is up and running?

Reply:
The webhook should definitely be started regardless of leadership, because it's stateless and having multiple replicas makes it more reliable. I think the metrics thread should probably only start after winning the election, but if the webhook starts producing metrics in the future it'd have to change back to this order.

Reply:
Makes sense.

Reply:
Okay, I'll reorder, and since the metric config will need to be passed into the webhook if it ever starts producing metrics, it'll be really obvious that the ordering needs to be changed again.